src/storage/iceberg_insert.cpp

```diff
@@ -159,8 +179,8 @@ static void AddWrittenFiles(IcebergInsertGlobalState &global_state, DataChunk &c
 		data_file.status = IcebergManifestEntryStatusType::ADDED;
 		data_file.file_format = "parquet";
 
-		if (partition_id.IsValid()) {
-			data_file.partition_spec_id = static_cast<int32_t>(partition_id.GetIndex());
+		if (partition_id) {
```

Comment:
This if/else is pretty much redundant now
But I am weary of using `ic_table.table_info.table_metadata.default_spec_id` here, because that is taken from the `rest_api_objects::TableMetadata`, which has an accompanying `has_default_spec_id`, which I'm not sure is checked.

Though that bool could only be used for stage-create, when a spec_id is not assigned to the table yet ?
Not sure